website/static/js/components/logFeed.js

```diff
@@ -0,0 +1,243 @@
+'use strict';
+
+require('css/log-feed.css');
+var $ = require('jquery');  // jQuery
+var m = require('mithril'); // exposes mithril methods, useful for redraw etc.
+var oop = require('js/oop');
+var $osf = require('js/osfHelpers');
+var mHelpers = require('js/mithrilHelpers');
+var Raven = require('raven-js');
+var LogText = require('js/logTextParser');
+var Paginator = require('js/paginator');
+
+var MAX_PAGES_ON_PAGINATOR = 7;
+var MAX_PAGES_ON_PAGINATOR_SIDE = 5;
+var LOG_PAGE_SIZE_LIMITED = 3;
+var LOG_PAGE_SIZE = 6;
+var PROFILE_IMAGE_SIZE = 16;
+
+var _buildLogUrl = function(node, page, limitLogs) {
+    var logPage = page || 1;
+    var urlPrefix = (node.isRegistration || node.is_registration) ? 'registrations' : 'nodes';
+    var size = limitLogs ? LOG_PAGE_SIZE_LIMITED : LOG_PAGE_SIZE;
+    var query = { 'page[size]': size, 'page': logPage, 'embed': ['original_node', 'user', 'linked_node', 'template_node'], 'profile_image_size': PROFILE_IMAGE_SIZE};
+    if (node.link) {
+        query.view_only = node.link;
+    }
+    return $osf.apiV2Url(urlPrefix + '/' + node.id + '/logs/', { query: query});
+};
+
+var LogFeed = {
+
+    controller: function(options) {
+        var self = this;
+        self.node = options.node;
+        self.activityLogs = m.prop();
+        self.logRequestPending = m.prop();
+        self.limitLogs = options.limitLogs;
+        self.failed = false;
+        self.paginators = m.prop([]);
+        self.nextPage = m.prop();
+        self.prevPage = m.prop();
+        self.firstPage = 0;
+        self.lastPage = 0;
+        self.totalPages = 0;
+        self.currentPage = m.prop();
+        self.pageToGet = m.prop();
+
+        self.getLogs = function _getLogs (url) {
+            self.activityLogs([]); // Empty logs from other projects while load is happening;
+            function _processResults (result){
+                result.data.map(function(log){
+                    log.attributes.formattableDate = new $osf.FormattableDate(log.attributes.date);
+                });
+
+                self.activityLogs(result.data);  // Set activity log data
+                self.nextPage(result.links.next);
+                self.prevPage(result.links.prev);
+                self.firstPage = result.links.first;
```

Comment:
Sorry to contradict what we had talked about in person, but `firstPage` and `lastPage` should actually be m.props because they do change over time. They are initialized in the controller, then they change after the first log request is made.

Making them props ensures that the view will update when they are updated. This is working now  only because `nextPage` and `prevPage` are also updated in the same callback.